internal/errors.go

```diff
@@ -0,0 +1,80 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"go.temporal.io/sdk/temporal"
+)
+
+// errorType identifies a custom Cadence error as an AWS SDK error.
+const errorType = "AWS_SDK_ERROR"
+
+// EncodedError is a serialized version of the AWS SDK error.
+type EncodedError struct {
+	Code    string
+	Message string
+	OrigErr string
```

Comment:
I would remove it as origError will be chained down there.

Reply:
Orig error can actually be another AWSError, which should also be properly encoded. That's one of the todo items in the PR. Not sure what would be the best way for that (yet).

Reply:
I would call this method on orig recursively.